### PR TITLE
Add documentation on templates features

### LINKED ISSUE #83: Add GIF of usage
Would be cool for demo purposes if we had a 10 second GIF showing the usage of this tool.

### PR DESCRIPTION
**Pull request by jorgepiloto**
This PR introduces the following enhancements to documentation: 

- [x] Exposes the capabilities and features of each template.
- [x] Improves the README.rst file by adding a GIF (resolves #83), installation guide, basic commands and a comparison table.

Since the link for the GIF in the README.rst is pointing to the main branch, it will not be visible until this gets merged. For a quick reference, see:

GIF usage

### CODE REVIEW

**Comment by jorgepiloto** [comment]
This is related to https://github.com/pyansys/ansys-templates/issues/84, although does not solve it.

**Comment by RobPasMue** [comment]
GIF looks just fine @jorgepiloto! =) I'll review the docs.

File: doc/source/templates.rst [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+Main features of this package are:
+
+- Does not use Ansys namespace, as it generates a  ``src/library`` layout.
+- Uses a ``setup.py`` file for project configuration.
+- Includes a ``pyrpoject.toml`` for tools configuration.
+- Includes ``doc/`` and ``tests/``.
+- No GitHub workflows (actions).
+- All ``requirements_*.txt`` are contained in the base directory.
+- No ``tox.ini`` file.
+
```
[the remark this answers is not in the document]
Comment (jorgepiloto) [reply doc/source/templates.rst:53]: I see your point... Gonna go for it 🚀

File: README.rst [hunk 2]
```diff
@@ -47,29 +47,94 @@ For information on using this tool, see the `Ansys Templates Documentation`_.
...
+.. code-block:: text
+        
+    python -m pip install ansys-templates
+
+The usage of `pipx`_ is encouraged too. See `installing ansys-templates using
+pipx`_.
+
+.. _pipx: https://pypa.github.io/pipx/
+.. _installing ansys-templates using pipx: https://templates.pyansys.com/getting_started/index.html#installing-pipx
+
```
[the remark this answers is not in the document]
Comment (jorgepiloto) [reply README.rst:65]: This works when rendering in Sphinx but not in the main page of the project.

**Review by RobPasMue — APPROVED** [review]

**Pull request merged**